File: backend/internal/models/notifications.go
package models

import (
	"encoding/json"
	"time"
)

type NotificationType string

const (
	TypeTaskAssigned     NotificationType = "task_assigned"
	TypeHighPriorityTask NotificationType = "high_priority_task"
)

type Notification struct {
	ID        string           `json:"id"`
	UserID    string           `json:"user_id"`
	Type      NotificationType `json:"type"`
	Title     string           `json:"title"`
	Body      string           `json:"body"`
	Data      json.RawMessage  `json:"data,omitempty"`
	ReadAt    *time.Time       `json:"read_at,omitempty"`
	CreatedAt time.Time        `json:"created_at"`
} //@name Notification

type RegisterDeviceTokenInput struct {
	Token    string `json:"token" validate:"notblank"`
	Platform string `json:"platform" validate:"oneof=ios android"`
} //@name RegisterDeviceTokenInput

File: clients/shared/src/types/notifications.ts
export const NotificationType = {
  TaskAssigned: "task_assigned",
  HighPriorityTask: "high_priority_task",
} as const;

export type NotificationType = (typeof NotificationType)[keyof typeof NotificationType];

export const DevicePlatform = {
  iOS: "ios",
  Android: "android",
} as const;

export type DevicePlatform = (typeof DevicePlatform)[keyof typeof DevicePlatform];

export type Notification = {
  id: string;
  user_id: string;
  type: NotificationType;
  title: string;
  body: string;
  data?: Record<string, unknown>;
  read_at?: string;
  created_at: string;
};

export type RegisterDeviceTokenInput = {
  token: string;
  platform: DevicePlatform;
};
